packages/ag-charts-enterprise/src/series/radial-bar/radialBarSeries.ts

```diff
@@ -0,0 +1,708 @@
+        if (direction === ChartAxisDirection.X) {
+            const angleAxis = axes[ChartAxisDirection.X];
+            const xExtent = dataModel.getDomain(this, 'angleValue-end', 'value', processedData);
+            const fixedXExtent = [xExtent[0] > 0 ? 0 : xExtent[0], xExtent[1] < 0 ? 0 : xExtent[1]];
+            return this.fixNumericExtent(fixedXExtent as any, angleAxis);
+        } else {
+            return dataModel.getDomain(this, 'radiusValue', 'key', processedData);
+        }
+    }
+
+    async processData(dataController: _ModuleSupport.DataController) {
+        const { data = [], visible } = this;
+        const { angleKey, radiusKey } = this;
+
+        if (!angleKey || !radiusKey) return;
+
+        const stackGroupId = this.getStackId();
+        const stackGroupTrailingId = `${stackGroupId}-trailing`;
+
+        const normalizedToAbs = Math.abs(this.normalizedTo ?? NaN);
+        const normaliseTo = normalizedToAbs && isFinite(normalizedToAbs) ? normalizedToAbs : undefined;
+        const extraProps = [];
+        if (normaliseTo) {
+            extraProps.push(normaliseGroupTo(this, [stackGroupId, stackGroupTrailingId], normaliseTo, 'range'));
+        }
+
+        const { dataModel, processedData } = await dataController.request<any, any, true>(this.id, data, {
+            props: [
+                keyProperty(this, radiusKey, false, { id: 'radiusValue' }),
+                valueProperty(this, angleKey, true, { id: 'angleValue-raw', invalidValue: undefined }),
+                ...groupAccumulativeValueProperty(this, angleKey, true, 'window', 'current', {
+                    id: `angleValue-end`,
+                    invalidValue: null,
+                    groupId: stackGroupId,
+                }),
+                ...groupAccumulativeValueProperty(this, angleKey, true, 'window-trailing', 'current', {
+                    id: `angleValue-start`,
+                    invalidValue: null,
+                    groupId: stackGroupTrailingId,
+                }),
+                ...extraProps,
+            ],
+            dataVisible: visible,
+        });
+
+        this.dataModel = dataModel;
+        this.processedData = processedData;
+    }
+
+    protected circleCache = { r: 0, cx: 0, cy: 0 };
+
+    protected didCircleChange() {
+        const r = this.radius;
+        const cx = this.centerX;
+        const cy = this.centerY;
+        const cache = this.circleCache;
+        if (!(r === cache.r && cx === cache.cx && cy === cache.cy)) {
+            this.circleCache = { r, cx, cy };
+            return true;
+        }
+        return false;
+    }
+
+    maybeRefreshNodeData() {
+        const circleChanged = this.didCircleChange();
+        if (!circleChanged && !this.nodeDataRefresh) return;
+        const [{ nodeData = [] } = {}] = this.createNodeDataSync();
+        this.nodeData = nodeData;
+        this.nodeDataRefresh = false;
+    }
+
+    async createNodeData() {
+        return this.createNodeDataSync();
```

Comment:
This seems completely unnecessary? Please maintain the async contract - all callers (specifically `maybeRefreshNodeData()`) are called by `async` functions so there should be no need for a synchronous version of this function.

Reply:
I saw it was implemented in Pie series that way. I will try making it async.

Reply:
Unfortunately there is `computeLabelsBBox` method of `PolarSeries` that is `sync`. Maybe I could make a follow up PR to fix it for all the polar series?

Reply:
Ok sounds good